interpolate.go

```diff
@@ -82,6 +82,19 @@ func (e UnsetValueExpansion) Expand(env Env) (string, error) {
 	return val, nil
 }
 
+// EscapedEspansion is an expansion that is delayed until later on (usually by a later process)
+type EscapedExpansion struct {
+	Identifier string
+}
+
+func (e EscapedExpansion) Identifiers() []string {
+	return []string{"$" + e.Identifier}
+}
+
+func (e EscapedExpansion) Expand(_ Env) (string, error) {
```

Comment:
No `_` needed if all args have no name:
```suggestion
func (e EscapedExpansion) Expand(Env) (string, error) {
```